trl/trainer/moreh_utils.py

```diff
@@ -11,17 +11,13 @@
 class TBTrainerCallback(TrainerCallback):
     "A callback log loss, learning rate, and throughput each logging step"
     start_time = time.time()
-    epoch_loss_list = []
+    epoch_start = 0
 
     def on_step_begin(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
         # count the time after the logging step
         if state.global_step == 0 or state.global_step % args.logging_steps == 1:
             self.start_time = time.time()
 
-    def on_step_end(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
```

Comment:
Just out of curiosity, does this mean we won't use average train loss when we use `steps` for logging strategy?

Reply:
I don't understand the question, can you explain more?

Reply:
I'm wondering why this `on_step_end` method was deleted in the latest commit. I just wanted to double check that you won't need this method since we are going to measure the average training loss only when the logging_strategy is set `epoch`, rather than `steps`.

Reply:
@yeomni ah I see. In the old code, I use on_step_end to collect the loss after each step but in the new code, I collect the loss directly from the state_history so this method is redundant.
The logging method is still steps, not epoch. I add a new function to collect the epoch log in addition to each step's logging.